plugins/auth-backend/src/providers/gcp-iap/helpers.ts

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { AuthenticationError } from '@backstage/errors';
+import { OAuth2Client, TokenPayload } from 'google-auth-library';
+import { AuthHandler } from '../types';
+import { GcpIapResult, IAP_JWT_HEADER } from './types';
+
+export function createTokenValidator(
+  audience: string,
+  mockClient?: OAuth2Client,
+): (token: string) => Promise<TokenPayload> {
+  return async function tokenValidator(token) {
+    const client = mockClient ?? new OAuth2Client();
```

Comment:
Yeah honestly this was only defensively because it was made on each sign-in in the code before me. It would be pretty crazy if there were tangible problems with holding one client for a longer time though. So I agree, let's move it out and only do something like this if it's deemed necessary for some buggy reason.